safe_transaction_service/notifications/utils.py

```diff
@@ -1,10 +1,65 @@
-from typing import Sequence
+import hashlib
+import json
+from typing import Any, Dict, Optional, Sequence
 from uuid import UUID
 
+from django.conf import settings
+
 from hexbytes import HexBytes
 
 from gnosis.eth.utils import fast_keccak
 
+from safe_transaction_service.utils.redis import get_redis
+
+
+class SafeNotification:
+    def __init__(self, address: Optional[str], payload: Dict[str, Any]):
+        self.redis = get_redis()
+        self.address = address
+        self.payload = payload
+        self.redis_key = self._get_redis_key(address, payload)
+        self.expiration_time_seconds = (
+            settings.NOTIFICATIONS_DUPLICATED_EXPIRATION_TIME_SECONDS
+        )
+
+    def _get_redis_key(self, address: Optional[str], payload: Dict[str, Any]) -> str:
+        """
+        :param address:
+        :param payload:
+        :return: Key built from ``address`` and MD5 hashing the ``payload``
+        """
+        payload_hash = json.dumps(payload, sort_keys=True)
+        # Use MD5 as it's fast and should be enough for this use case
+        hex_hash = hashlib.md5(payload_hash.encode()).hexdigest()
+        return f"notifications:{address}:{hex_hash}"
+
+    def is_duplicated(self) -> bool:
```

Comment:
Looks like the last changes this function is used only for testing, do you think that is still necessary?

Reply:
Even if it's only for testing, the function can be useful in further iterations of this feature, so I wouldn't remove it